hashring/hashring_test.go

```diff
@@ -177,6 +177,80 @@ func TestLookup(t *testing.T) {
 	assert.False(t, ok, "expected Lookup to find no server for key to hash to")
 }
 
+func TestLookupDistribution(t *testing.T) {
+	ring := New(farm.Fingerprint32, 5)
+	addresses := genAddresses(1, 1, 1000)
+	ring.AddRemoveServers(addresses, nil)
+
+	keys := make([]string, 40)
+	for i := range keys {
+		keys[i] = fmt.Sprintf("%d", i)
+	}
+
+	servers := make(map[string]bool)
+	for _, key := range keys {
+		server, ok := ring.Lookup(key)
+		assert.True(t, ok, "expected that the lookup is a success")
+		servers[server] = true
+	}
+
+	// Expect no more than one colission
+	assert.Len(t, servers, len(keys), "expected that all keys are owned by a different server")
+}
+
+// TestLookupNNoGaps tests the selected servers from lookup N are
```

Comment:
"tests the selected servers from lookup N are form" => "tests that the servers selected by lookupN form"